src/main/java/org/broadinstitute/hellbender/utils/diffengine/BAMDiffableReader.java

```diff
@@ -0,0 +1,82 @@
+package org.broadinstitute.hellbender.utils.diffengine;
+
+import htsjdk.samtools.*;
+import htsjdk.samtools.util.BlockCompressedInputStream;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+
+import java.io.*;
+import java.util.Arrays;
+
+/**
+ * Class implementing diffnode reader for BAM/SAM/CRAM files.
+ */
+final class BAMDiffableReader implements DiffableReader {
+
+    private static final byte[] BAM_MAGIC = "BAM\1".getBytes();
+
+    @Override
+    public String getName() { return "BAM"; }
+
+    @Override
+    public DiffElement readFromFile(final File file, final int maxElementsToRead) throws IOException{
+        try(final SamReader reader = SamReaderFactory.makeDefault().validationStringency(ValidationStringency.SILENT).open(file)) {
```

Comment:
Will not work with cram files due to lack of a reference passed into the factory (see GATKTool.initializeReads()). Should we worry about that now?